src/backend/cdb/cdbllize.c

```diff
@@ -659,6 +659,20 @@ ParallelizeCorrelatedSubPlanMutator(Node *node, ParallelizeCorrelatedPlanWalkerC
 		return ParallelizeCorrelatedSubPlanMutator(node, ctx);
 	}
 
+	if (IsA(node, ModifyTable))
+	{
+		Plan   *mplan = (Plan *) node;
+		if (ctx->movement == MOVEMENT_BROADCAST)
+			broadcastPlan(mplan, false /* stable */ , false /* rescannable */,
+					ctx->currentPlanFlow->numsegments);
+		else
+			focusPlan(mplan, false /* stable */ , false /* rescannable */);
+
+		Plan   *mat = materialize_subplan((PlannerInfo *) ctx->base.node, mplan);
```

Comment:
Ready to approve, but we should move declaration of this variable upper, according to conventions.
The block above (you inspired by) has logical division to sub-blocks, which is also breaks convention, but anyway.

Reply:
```c
return (Node *) materialize_subplan((PlannerInfo *) ctx->base.node, mplan);
```

Reply:
done